examples/seismic/elastic/operators.py

```diff
@@ -26,56 +83,111 @@ def ForwardOperator(model, source, receiver, space_order=4,
+    u_txy = Eq(txy.forward, damp * txy - damp * mu * s * (vy.forward.dx + vx.forward.dy))
+    u_tyz = Eq(tyz.forward, damp * tyz - damp * mu * s * (vy.forward.dz + vz.forward.dy))
+
+    src_rec_expr = src_rec(vx, vy, vz, txx, tyy, tzz, model, source, receiver)
+    return [u_vx, u_vy, u_vz, u_txx, u_tyy, u_tzz, u_txz, u_txy, u_tyz] + src_rec_expr
+
+
+def src_rec(vx, vy, vz, txx, tyy, tzz, model, source, receiver):
+    """
+    Source injection and receiver interpolation
+    """
+    s = model.grid.time_dim.spacing
     # Source symbol with input wavelet
     src = PointSource(name='src', grid=model.grid, time_range=source.time_range,
                       npoint=source.npoint)
     rec1 = Receiver(name='rec1', grid=model.grid, time_range=receiver.time_range,
                     npoint=receiver.npoint)
     rec2 = Receiver(name='rec2', grid=model.grid, time_range=receiver.time_range,
                     npoint=receiver.npoint)
-    # Stencils
-    fd_vx = (staggered_diff(txx, dim=x, order=space_order, stagger=left) +
-             staggered_diff(txz, dim=z, order=space_order, stagger=right))
-    u_vx = Eq(vx.forward, damp * vx - damp * s * ro * fd_vx)
-
-    fd_vz = (staggered_diff(txz, dim=x, order=space_order, stagger=right) +
-             staggered_diff(tzz, dim=z, order=space_order, stagger=left))
-    u_vz = Eq(vz.forward, damp * vz - damp * ro * s * fd_vz)
-
-    vxdx = staggered_diff(vx.forward, dim=x, order=space_order, stagger=right)
-    vzdz = staggered_diff(vz.forward, dim=z, order=space_order, stagger=right)
-    u_txx = Eq(txx.forward, damp * txx - damp * (l + 2 * mu) * s * vxdx
-                                       - damp * l * s * vzdz)
-    u_tzz = Eq(tzz.forward, damp * tzz - damp * (l+2*mu)*s * vzdz
-                                       - damp * l * s * vxdx)
-
-    vxdz = staggered_diff(vx.forward, dim=z, order=space_order, stagger=left)
-    vzdx = staggered_diff(vz.forward, dim=x, order=space_order, stagger=left)
-    u_txz = Eq(txz.forward, damp * txz - damp * mu*s * (vxdz + vzdx))
 
     # The source injection term
     src_xx = src.inject(field=txx.forward, expr=src * s, offset=model.nbpml)
     src_zz = src.inject(field=tzz.forward, expr=src * s, offset=model.nbpml)
+    src_expr = src_xx + src_zz
+    if model.grid.dim == 3:
+        src_yy = src.inject(field=tyy.forward, expr=src * s, offset=model.nbpml)
+        src_expr += src_yy
 
     # Create interpolation expression for receivers
-    rec_term1 = rec1.interpolate(expr=txx, offset=model.nbpml)
-    rec_term2 = rec2.interpolate(expr=tzz, offset=model.nbpml)
+    rec_term1 = rec1.interpolate(expr=tzz, offset=model.nbpml)
+    if model.grid.dim == 2:
+        rec_expr = vx.dx + vz.dy
+    else:
+        rec_expr = vx.dx + vy.dy + vz.dz
+    rec_term2 = rec2.interpolate(expr=rec_expr, offset=model.nbpml)
+
+    return src_expr + rec_term1 + rec_term2
+
+
+def ForwardOperator(model, source, receiver, space_order=4,
```

Comment:
remind me why we don't have a class like we have instead for acoustic and tti?

Reply:
Exact same setup as acoustic and tti

Reply:
I thought we have a class in the other examples, in which ``ForwardOperator`` acted as a method. If it's a free function, then we should rather call it ``forward_operator`` (Python -- and not just Python tbh -- convention)